packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -190,13 +217,13 @@ const devHtmlHook: IndexHtmlTransformHook = async (
 
   await Promise.all(
     styleUrl.map(async ({ start, end, code }, index) => {
-      const url = filename + `?html-proxy&${index}.css`
+      const url = `${proxyModulePath}?html-proxy&index=${index}.css`
```

Comment:
The url was missing `index=` here, I think it worked because this is only needed as a temporary name but still good to follow the same convention as with JS I think.

Reply:
BTW, `mod.id` is `/path-to/index.html?html-proxy&index=0.css` 😀